waku-org/nwaku
# chore(keystore): verbose error message when credential is not found

# Description

This PR adds some verbose logging when a keystore has multiple credentials, but the `rlnRelayCredIndex` is not passed in to find the unique credential. The error log now has some indication of what should be done to fix it, rather than just saying `credential not found`.

# Changes



- [x] logging change in `waku_keystore.KeystoreCredentialNotFoundError`

## How to test

1. Generate a keystore with multiple credentials by using `rlnKeystoreGenerator` multiple times for the same keystore
2. Run `wakunode2`:  `./build/wakunode2 --cluster-id:1 --rln-relay-cred-path:rlnKeystore.json --rln-relay-eth-client-address:https://eth-sepolia.g.alchemy.com/v2/<YOUR_API_KEY> --rln-relay-cred-password:password`
3. The following error log is shown:
<img width="1130" alt="image" src="https://github.com/user-attachments/assets/b13dccb6-b2e4-4ffd-932a-17f54a8fc781">
